cmd/kubeadm/app/phases/upgrade/health.go

```diff
@@ -139,41 +140,18 @@ func createJob(client clientset.Interface, cfg *kubeadmapi.ClusterConfiguration)
 		},
 	}
 
-	// Check if the Job already exists and delete it
-	if _, err := client.BatchV1().Jobs(ns).Get(context.TODO(), jobName, metav1.GetOptions{}); err == nil {
-		if err = deleteHealthCheckJob(client, ns, jobName); err != nil {
-			return err
-		}
-	}
-
-	// Cleanup the Job on exit
-	defer func() {
-		lastError = deleteHealthCheckJob(client, ns, jobName)
-	}()
-
-	// Create the Job, but retry in case it is being currently deleted
-	klog.V(2).Infof("Creating Job %q in the namespace %q", jobName, ns)
-	err := wait.PollImmediate(time.Second*1, timeout, func() (bool, error) {
-		if _, err := client.BatchV1().Jobs(ns).Create(context.TODO(), job, metav1.CreateOptions{}); err != nil {
-			klog.V(2).Infof("Could not create Job %q in the namespace %q, retrying: %v", jobName, ns, err)
-			lastError = err
-			return false, nil
-		}
-		return true, nil
-	})
+	ctx := context.Background()
+	createdJob, err := client.BatchV1().Jobs(ns).Create(ctx, job, metav1.CreateOptions{})
```

Comment:
the CREATE API call should still be polled in a PollUntilContextTimeout

Reply:
@neolit123 what's the reason why need retry when we create a job with a GenerateName? 

```
// Create the Job, but retry in case it is being currently deleted
klog.V(2).Infof("Creating Job %q in the namespace %q", jobName, ns)
err := wait.PollImmediate(time.Second*1, timeout, func() (bool, error) {
if _, err := client.BatchV1().Jobs(ns).Create(context.TODO(), job, metav1.CreateOptions{}); err != nil {
	klog.V(2).Infof("Could not create Job %q in the namespace %q, retrying: %v", jobName, ns, err)
	lastError = err
	return false, nil
}
return true, nil
})
```

Reply:
users sometimes report that their API server is not responding for small amounts of time.
it's better to have retry for all API calls from kubeadm.